crates/fs/private/node.rs

```diff
@@ -0,0 +1,280 @@
+use std::{io::Cursor, rc::Rc};
+
+use anyhow::{anyhow, bail, Result};
+use chrono::{DateTime, Utc};
+use libipld::{
+    cbor::DagCborCodec,
+    codec::{Decode, Encode},
+    serde as ipld_serde, Ipld,
+};
+use serde::{Deserialize, Serialize, Serializer};
+use sha3::Sha3_256;
+use skip_ratchet::Ratchet;
+
+use crate::{FsError, HashOutput, Id, Metadata};
+
+use super::{
+    hamt::Hasher, namefilter::Namefilter, Key, PrivateDirectory, PrivateDirectoryContent,
+    PrivateFile, PrivateFileContent,
+};
+
+//--------------------------------------------------------------------------------------------------
+// Type Definitions
+//--------------------------------------------------------------------------------------------------
+
+pub type INumber = HashOutput;
+
+#[derive(Debug, PartialEq, Eq, Clone, Serialize, Deserialize)]
+pub struct ContentKey(pub Key);
+
+#[derive(Debug, PartialEq, Eq, Clone, Serialize, Deserialize)]
+pub struct RatchetKey(pub Key);
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct PrivateNodeHeader {
+    pub(crate) bare_name: Namefilter,
+    pub(crate) ratchet: Ratchet,
+    pub(crate) inumber: INumber,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum PrivateNode {
+    File(Rc<PrivateFile>),
+    Dir(Rc<PrivateDirectory>),
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct PrivateRef {
+    pub(crate) saturated_name_hash: HashOutput, // Sha3-256 hash of saturated namefilter
+    pub(crate) content_key: ContentKey,         // A hash of ratchet key.
+    pub(crate) ratchet_key: RatchetKey,         // Encrypted ratchet key.
```

Comment:
I think this comment is stale now, it's not the "Encrypted" ratchet key anymore.